### DISCUSSION TITLE
Cppfront primitive types

### DISCUSSION THREAD

**Discussion by lauler1** [hsutter/cppfront#866] [Q&A]
I know that the C++ primitive types are discouraged, although the cppfront equivalents are declared in `cpp2util.h`, e.g.:

```cpp
using ushort     = unsigned short;
using uint       = unsigned int;
using ulong      = unsigned long;
using longlong   = long long;
using ulonglong  = unsigned long long;
using longdouble = long double;
using _schar     = signed char;      // normally use i8 instead
using _uchar     = unsigned char;    // normally use u8 instead
```

However, I noticed that not all of them work as expected. For example:

- With `ushort`, I get `cpp2::ushort`: **OK**
- With `ulong`, I get `cpp2::ulong`: **OK**
- With `longlong`, I get `cpp2::longlong`: **OK**

But:

- With `ulonglong`, I get `ulonglong`: **Not OK**
- With `_schar`, I get `_schar`: **Not OK**
- With `_uchar`, I get `_uchar`: **Not OK**

Why is this happening?

**Comment by hsutter** [comment]
Thanks! That's a bug, fixed in 089a534d7e4515230eb35baa952a769a47058e05
